runtime/move_float.ll

```diff
@@ -1,16 +1,23 @@
 target datalayout = "e-m:e-i64:64-f80:128-n8:16:32:64-S128"
 target triple = "x86_64-unknown-linux-gnu"
 
-%floating = type { i64, { i64, i32, i64, i64 * } } ; exp, mpfr_t
+%blockheader = type { i64 }
+%floating = type { %blockheader, i64, { i64, i32, i64, i64 * } } ; 16-bit layout, 48-bit length, exp, mpfr_t
 
 ; helper function for float hooks
 define %floating* @move_float(%floating* %val) {
-  %loaded = load %floating, %floating* %val
-  %malloccall = tail call i8* @koreAllocNoGC(i64 ptrtoint (%floating* getelementptr (%floating, %floating* null, i32 1) to i64))
+  %p1 = getelementptr inbounds %floating, %floating* %val, i64 0, i32 1
```

Comment:
why has this function gotten more complicated? I would have thought you just needed to change the allocation function.

Reply:
Now that `floating` has a block header, if we just changed the allocation function to `koreAllocFloating` and then copied the whole floating object on the allocated space, the initialization of the header performed by `koreAllocFloating` would be undone. So I had to either copy only the non-header part or reinitialize the header. I thought the latter is duplicating the logic of header initialization, so I chose the former approach.

Reply:
I now understand what is going on here, but I think I have decided I don't like what this does to the code asymmetry to treat floats differently than ints. Is it possible we could have a nested struct containing just the exponent and mpfr_t and treat that nested struct the same way we treat mpz_t?

Reply:
Yes I don't see why not. It would require a lot of changes in the implementation of float hooks and elsewhere, although mostly refactoring changes. I can do if you are OK with that.

Reply:
yeah, I think that's probably the right approach. Either that or to make blocks contain a integer * instead of an mpz_ptr. Personally I like the former better because it means that the hooks don't need to worry about block headers.

Reply:
I implemented the discussed refactoring. Please take a look again when you can.